GOVERNANCE.md

```diff
@@ -0,0 +1,226 @@
+- handling code of conduct violations
+- managing financial decisions
+- defining the scope and responsibilities of all other teams
+- making descisions that affect multiple teams
+- resolving escalated team decisions when the teams responsible are blocked
+- changes to governance (this document)
+
+### Dev team
+
+@fluxcd/dev <https://github.com/orgs/fluxcd/teams/dev>
+
+Note: Dev team members are scoped to one or more source code repositories, where they will be listed in a MAINTAINERS file in the root of that repo.
+How current Dev team members are allocated to which git repos may be decided by a Simple Majority of existing members listed in a git repo's MAINTAINERS file.
+In future, if more useful than relying on MAINTAINERS files, this team may explicitly split into separate dev-related teams, each scoped to one or more git repos.
+
+Scope and responsibilities:
+
+- shared Maintainers responsibilities, as well as:
+- all technical decisions related to project software, according to member's scoped git repos
+- release management, according to member's scoped git repos
+
+### Security team
+
+@fluxcd/security <https://github.com/orgs/fluxcd/teams/security>
+
+Scope and responsibilities:
+
+- shared Maintainers responsibilities, as well as:
+- point of contact for reporting security vulnerabilities
+- responding to and investigating security reports
+- patching and releasing security fixes
+- publishing the vulnerability disclosure once a mitigation strategy is available
+- documenting overall project security best practices
+
+Anyone may email this team privately to safely raise possible security vulnerabilities <https://lists.cncf.io/g/cncf-flux-community>.
+
+### Community team
+
+@fluxcd/community <https://github.com/orgs/fluxcd/teams/community>
+
+Scope and responsibilities:
+
+- shared Maintainers responsibilities, as well as
+- listening to the community
+- escalating community issues to appropriate teams
+- planning and executing community events (example: GitOps Days)
+- marketing related tasks required by the project
+- managing partner and ecosystem development
+- managing social media accounts (example: Twitter)
+- managing project mailing lists
+- managing project calendar events
+- managing project blog posts
+- managing the project website
+
+Anyone may email this team with community-related questions <https://lists.cncf.io/g/cncf-flux-community>.
+
+## Proposal Process
+
+- Code changes should go through the pull request process, where the idea and implementation details can be publicly discussed with maintainers of the appropriate team, other contributors, and end users.
+Pull requests should only be merged after receiving GitHub approval from at least one other member of the appropriate team.
+Voting is not required for most code changes.
+Note that Flux v2 uses GitHub discussions for proposals in the `fluxcd/toolkit` repo <https://github.com/fluxcd/toolkit/discussions?discussions_q=category%3AProposals>.
```

Comment:
Should there be any criteria for when a proposal needs a github discussion vs a direct pull request? Perhaps something related to semver e.g. anything which is not a simple PATCH (backward compatible bugfix) change requires a github discussion. Then someone can watch the github discussions to keep up vs needing to watch all repos they care about.

Reply:
I would say any change to an existing API that affects end-users should be discussed first, same for new APIs before creating the RFC pull request on a particular repo.

Reply:
@stefanprodan that sounds right. Do you think we can leave that to be defined by the Architecture team in a separate contribution README, or must we include that detail in this governance document?

Reply:
We could have a dedicated doc on this 👍